### DISCUSSION TITLE
input_box.py example bug?

### DISCUSSION THREAD

**Discussion by sergei9838** [marcomusy/vedo#1044] [Q&A] [answered]
Hi, Marco,
`examples/basic/input_box.py` is not working as it should. The text in the box does not redraw when I correct symbols with Backspace. Also, hitting Return closes the renderer window. I observe that, at least on my system MacOS Monterey, the erase key is not called `BackSpace`, but `Backspace`. Finally, adding `interactive=True` keeps the renderer open and ready for other inputs. Below is the code that works on my system (two lines altered)

```
"""Start typing a color name then press return
E.g. pink4"""
from vedo import settings, dataurl, Plotter, Mesh

settings.enable_default_keyboard_callbacks = False


def kfunc(evt):
    global msg
    evt.keypress = evt.keypress.replace("period", ".")
    if evt.keypress == "Backspace" and msg:
        msg = msg[:-1]
        evt.keypress = ''
    elif evt.keypress == "Return":
        bfunc(0)
        return
    elif evt.keypress == "Escape":
        plt.close()

    if len(evt.keypress) > 1:
        return

    msg += f"{evt.keypress}"
    bu.text(msg)
    plt.render()


def bfunc(obj, ename=""):
    mesh.color(msg)
    plt.render()


plt = Plotter(axes=1)
plt.remove_callback("CharEvent") # might be needed

msg = ""
plt.add_callback("key press", kfunc)


bu = plt.add_button(
    bfunc,
    pos=(0.5, 0.1),  # x,y fraction from bottom left corner
    states=["input box"],
    c=["w"],
    bc=["dg"],        # colors of states
    font="courier",   # arial, courier, times
    size=45,
    bold=True,
)


mesh = Mesh(dataurl+"magnolia.vtk").c("v").flat()

plt.show(mesh, __doc__, interactive=True).close()

```

**Comment by marcomusy** [comment, marked as answer]
Thanks Sergei for drawing my attention to it, that needed a deep reno :)
just pushed a new version of it.

<img width="396" alt="Screenshot 2024-02-07 at 21 45 22" src="https://github.com/marcomusy/vedo/assets/32848391/d549aa2f-c527-4b5e-922c-bafc6deda362">
